pkg/instancetype/instancetype.go

```diff
@@ -959,6 +960,22 @@ func applyHostDevices(field *k8sfield.Path, instancetypeSpec *instancetypev1beta
 	return nil
 }
 
+func applyCPUPreferences(preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) {
+	if preferenceSpec.CPU == nil || len(preferenceSpec.CPU.PreferredCPUFeatures) == 0 {
+		return
+	}
+	// Only apply any preferred CPU features when the same feature has not been provided by a user already
+	cpuFeatureNames := make(map[string]*struct{})
```

Comment:
Does it make a difference using `*struct{}` or `struct{}`? I've seen this pattern used a lot with the non-ptr `struct{}`.

Reply:
Yup I forget why I went with a pointer here, I'll stick with the typical pattern thanks.